src/DetailView/EventDiagram.js

```diff
@@ -1,12 +1,117 @@
 import React, { Component } from 'react';
 import { css } from 'aphrodite';
+import plotly from 'plotly.js/dist/plotly.min';
+import Utils from './../Utils/Utils';
 
 class EventDiagram extends Component {
+
+	constructor(props) {
+		super(props);
+		this.diagramId = "eventDiagram";
+	}
+
+	static sortEventsByTime(eventA, eventB, timeStampAttributeName) {
+		const dateA = new Date(EventDiagram.getTimeStampAsStringFromEvent(eventA, timeStampAttributeName));
+		const dateB = new Date(EventDiagram.getTimeStampAsStringFromEvent(eventB, timeStampAttributeName));
+		return dateA - dateB;
+	}
+
+	static getTimeStampAsStringFromEvent(event, timeStampAttributeName) {
+		return event.Attributes.find(attribute =>
+			attribute.Name === timeStampAttributeName).Value;
+	}
+
+	static getDateAsStringFromEvent(event, timeStampAttributeName) {
+		return EventDiagram
+			.getTimeStampAsStringFromEvent(event, timeStampAttributeName)
+			.split("T")[0];
+	}
+
+	static sortEvents(events, timeStampAttributeName) {
+		return events.sort((eventA, eventB) => {
+			return EventDiagram.sortEventsByTime(eventA, eventB, timeStampAttributeName);
+		});
+	}
+
+	static getDiagramLayout(eventCounter) {
+		return {
+			yaxis: {
+				nticks: eventCounter + 2,
+			},
+			xaxis: {
+				showgrid: false,
+			},
+			height: 400
+		};
+	}
+
+	static getTrace(x, y, entity, eventCounter) {
+		const statusColor = Utils.getColorForStatus(entity.Status);
+		const opacityFactor = '80';
+		const traceData = {
+			x: x,
+			y: y,
+			type: 'line',
+			line: {
+				color: statusColor
+			}
+		};
+		if (eventCounter !== 1) {
+			traceData.fill = 'tozeroy';
+			traceData.fillcolor = statusColor + opacityFactor;
+		}
+		return [traceData];
+	}
+
+	componentWillReceiveProps(props) {
+		this.buildChartDataset(props);
+	}
+
+	buildChartDataset(props) {
+		const {events, eventType, eventTypeAttributes, entity} = props;
+		if (!events) {
+			return;
+		}
+
+		const x = [];
+		const y = [];
+		const dateCounter = {};
+		let eventCounter = 0;
+		const timeStampAttributeName = eventTypeAttributes.find(attribute =>
+			attribute.Id === eventType.TimestampAttributeId).Name;
+
+		const sortedEvents = EventDiagram.sortEvents(events, timeStampAttributeName);
+
+		//build x-axis
+		sortedEvents.forEach(event => {
+			eventCounter += 1;
+			const eventDate = EventDiagram.getDateAsStringFromEvent(event, timeStampAttributeName);
+			const currentDateCount = dateCounter[eventDate];
+			if (!currentDateCount) {
+				dateCounter[eventDate] = eventCounter;
+				x.push(eventDate);
+			}
+			else {
+				dateCounter[eventDate] = currentDateCount + 1;
+			}
+		});
+
+		//build y-axis
+		x.forEach(date =>
+			y.push(dateCounter[date]));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this logging statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS2228)